### PR TITLE
Case 19140: Clean up Luci ui to aggregate more tools

### PR DESCRIPTION
**Pull request by samcake**
https://highfidelity.fogbugz.com/f/cases/19140/Accessing-the-task-Job-Config-from-Scripting-interface-is-not-reliable-enough-in-qml-and-sometimes-crashes-let-s-fix-that

Just group a bunch of ui under the Luci ui

## TEST PLAN
Not needed, turst the developer

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14271/HighFidelity-Beta-PR14254-bb3d382.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14271/HighFidelity-Beta-PR14254-bb3d382.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14271/HighFidelity-Beta-Interface-PR14254-bb3d382.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14271/HighFidelity-Beta-PR14254-bb3d382.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14271/HighFidelity-Beta-Interface-PR14254-bb3d382.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14279/HighFidelity-Beta-PR14254-93f5e2d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14279/HighFidelity-Beta-PR14254-93f5e2d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14279/HighFidelity-Beta-Interface-PR14254-93f5e2d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14279/HighFidelity-Beta-PR14254-93f5e2d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14279/HighFidelity-Beta-Interface-PR14254-93f5e2d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14281/HighFidelity-Beta-PR14254-60447bc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14281/HighFidelity-Beta-PR14254-60447bc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14281/HighFidelity-Beta-Interface-PR14254-60447bc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14281/HighFidelity-Beta-PR14254-60447bc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14281/HighFidelity-Beta-Interface-PR14254-60447bc.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14285/HighFidelity-Beta-PR14254-702f622.apk

File: interface/src/ui/overlays/Overlay.cpp [hunk 1]
```diff
@@ -247,7 +247,7 @@ void Overlay::removeMaterial(graphics::MaterialPointer material, const std::stri
 }
 
 render::ItemKey Overlay::getKey() {
-    auto builder = render::ItemKey::Builder().withTypeShape();
+    auto builder = render::ItemKey::Builder().withTypeShape().withTypeMeta();
```
Comment (samcake) [inline interface/src/ui/overlays/Overlay.cpp]: THis is the fix for the tablet (and any OverlayModel) disappearing while moving around.
there meta render item was not tagged as meta hence not doing the culling from meta correctly

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14285/HighFidelity-Beta-PR14254-702f622.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14285/HighFidelity-Beta-Interface-PR14254-702f622.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14285/HighFidelity-Beta-PR14254-702f622.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14285/HighFidelity-Beta-Interface-PR14254-702f622.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14288/HighFidelity-Beta-PR14254-25213fe.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14288/HighFidelity-Beta-PR14254-25213fe.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14288/HighFidelity-Beta-Interface-PR14254-25213fe.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14288/HighFidelity-Beta-PR14254-25213fe.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14288/HighFidelity-Beta-Interface-PR14254-25213fe.exe

**Comment by qaAdamSmith82** [comment]
Fail @conklin94122 
The Tablet frame will still disappear if the user moves/looks away from it

Repro:
1. Launch PR 14254
2. In HMD mode bring up your tablet
3. Turn your head to the left and move with the right thumbstick
4. Look back at where you summoned your tablet
5. Observe frame is missing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14316/HighFidelity-Beta-PR14254-4565422.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14316/HighFidelity-Beta-PR14254-4565422.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14316/HighFidelity-Beta-Interface-PR14254-4565422.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14316/HighFidelity-Beta-PR14254-4565422.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14316/HighFidelity-Beta-Interface-PR14254-4565422.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14379/HighFidelity-Beta-PR14254-36f2b35.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14379/HighFidelity-Beta-PR14254-36f2b35.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14379/HighFidelity-Beta-Interface-PR14254-36f2b35.dmg

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14379/HighFidelity-Beta-Interface-PR14254-36f2b35.exe

File: scripts/developer/utilities/render/luci.js [hunk 2]
```diff
@@ -11,122 +11,131 @@
...
+    var onMousePressEvent = function (e) {
         if (e.isMiddleButton) {
             moveDebugCursor = true;
             setDebugCursor(e.x, e.y);
         }
-    });
-    Controller.mouseReleaseEvent.connect(function() { moveDebugCursor = false; });
-    Controller.mouseMoveEvent.connect(function (e) { if (moveDebugCursor) setDebugCursor(e.x, e.y); });
+    };
+    Controller.mousePressEvent.connect(onMousePressEvent);
```
Comment (ZappoMan) [inline scripts/developer/utilities/render/luci.js]: please make this a different PR... it's great, I will happily project approve Luci features in a different PR... but not this one. Much love! Thank you! Merci Beaucoup!

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14510/HighFidelity-Beta-PR14254-592c2b2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14510/HighFidelity-Beta-PR14254-592c2b2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14510/HighFidelity-Beta-Interface-PR14254-592c2b2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14510/HighFidelity-Beta-PR14254-592c2b2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14510/HighFidelity-Beta-Interface-PR14254-592c2b2.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14611/HighFidelity-Beta-PR14254-e5a96f2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14611/HighFidelity-Beta-PR14254-e5a96f2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14611/HighFidelity-Beta-Interface-PR14254-e5a96f2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14611/HighFidelity-Beta-PR14254-e5a96f2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14611/HighFidelity-Beta-Interface-PR14254-e5a96f2.exe

File: scripts/developer/utilities/render/luci.js (hunk 2 above)
Comment (samcake) [reply scripts/developer/utilities/render/luci.js]: hat s what i m doing with this pr, i removed any useful fix... :))))))))))))))))

**Comment by samcake** [comment]
THis PR was meant to Fix the Tablet bug (17284)
Since i was getting no where, it s now JUST improving Luci ui for us brave developers
Please merge asap

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14871/HighFidelity-Beta-PR14254-c40ded5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14871/HighFidelity-Beta-PR14254-c40ded5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14871/HighFidelity-Beta-Interface-PR14254-c40ded5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14871/HighFidelity-Beta-PR14254-c40ded5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14871/HighFidelity-Beta-Interface-PR14254-c40ded5.exe

**Pull request merged**